karabo_data/geometry2.py

```diff
@@ -116,7 +241,18 @@ def from_quad_positions(cls, quad_pos, asic_gap=2, panel_gap=29):
         The quadrant positions are given in pixel units, referring to the first
         pixel of the first module in each quadrant, corresponding to data
         channels 0, 4, 8 and 12.
+
+        The origin of the coordinates is in the centre of the detector.
+        Coordinates increase upwards and to the left (looking along the beam).
+
+        To give positions in units other than pixels, pass the *unit* parameter
+        as the length of the unit in metres.
+        E.g. ``unit=1e-3`` means the coordinates are in millimetres.
         """
+        px_conversion = unit / cls.pixel_size
+        asic_gap *= px_conversion
+        panel_gap *= px_conversion
+
         quads_x_orientation = [1, 1, -1, -1]
```

Comment:
Sorry to bring this again but in this layout 
https://github.com/European-XFEL/karabo_data/blob/lpd-geom2/docs/_static/agipd_layout.png 

x orientation of Quadrant 1 and 2 is in negative X direction. Or `quads_x_orientation` denotes something else here?

Reply:
Nevermind. I think I got it now. It is used later to specify slow and fast scan vectors
`ss_vec=np.array([x_orient, 0, 0]),`

Reply:
No, quadrants 1 and 2 are oriented in positive x direction - the first pixel is at the end with lower x, and then it reads across in increasing x direction. Remember that the diagrams all have x increasing right-to-left.